### PR TITLE
Task-37293: Ability to enable/disable the external invitation feature from administration

### PR DESCRIPTION
**Pull request by AzmiTouil**

### CODE REVIEW

File: webapp/portlet/src/main/resources/locale/portlet/social/SpacesAdministrationPortlet_en.properties [hunk 1]
```diff
@@ -21,6 +21,10 @@ social.spaces.administration.manageSpaces.space=Space
 social.spaces.administration.manageSpaces.description=Description
 social.spaces.administration.manageSpaces.visibility=Visibility
 
+social.spaces.administration.manageSpaces.inviteNewExternal=Invite new external
```
Comment (azayati) [inline webapp/portlet/src/main/resources/locale/portlet/social/SpacesAdministrationPortlet_en.properties]: Add FR labels also

File: webapp/portlet/src/main/webapp/spaces-administration/components/ExoSpacesAdministrationManageSpaces.vue [hunk 2]
```diff
@@ -162,7 +187,8 @@ export default {
       bindingsLoading: true,
       binding: {},
       groupPrettyName: '',
-      avatar : spacesConstants.DEFAULT_SPACE_AVATAR
+      avatar : spacesConstants.DEFAULT_SPACE_AVATAR,
+      ExternalFeatureEnabled: true
```
Comment (azayati) [inline webapp/portlet/src/main/webapp/spaces-administration/components/ExoSpacesAdministrationManageSpaces.vue]:
```suggestion
      externalFeatureEnabled: true
```

File: webapp/portlet/src/main/webapp/spaces-administration/spacesAdministrationServices.js [hunk 3]
```diff
@@ -113,3 +113,45 @@ export function getBindingReportOperations() {
...
+      Accept: 'application/json',
+      'Content-Type': 'application/json',
+    },
+  }).then((resp) => {
+    if (resp && resp.ok) {
+      return resp.json();
+    } else if (resp && resp.status === 404) {
+      return null;
+    } else {
+      throw new Error('Error External Feature settings');
```
Comment (azayati) [inline webapp/portlet/src/main/webapp/spaces-administration/spacesAdministrationServices.js]:
```suggestion
      throw new Error('Error when getting External Feature settings');
```

File: webapp/portlet/src/main/webapp/spaces-administration/spacesAdministrationServices.js [hunk 4]
```diff
@@ -113,3 +113,45 @@ export function getBindingReportOperations() {
...
+    }
+  }).then(resp => {
```
Comment (azayati) [inline webapp/portlet/src/main/webapp/spaces-administration/spacesAdministrationServices.js]: To be verified

File: webapp/portlet/src/main/webapp/spaces-administration/spacesAdministrationServices.js [hunk 5]
```diff
@@ -113,3 +113,45 @@ export function getBindingReportOperations() {
...
+    body: JSON.stringify({
+      value: !status,
+    }),
+  }).then((resp) => {
+    if (resp && resp.ok) {
+      return resp;
+    } else {
+      throw new Error('Error setting External Feature settings');
+    }
+  }).then(resp => {
```
Comment (azayati) [inline webapp/portlet/src/main/webapp/spaces-administration/spacesAdministrationServices.js]: To be verified

File: webapp/portlet/src/main/webapp/spaces-administration/spacesAdministrationServices.js [hunk 6]
```diff
@@ -113,3 +113,45 @@ export function getBindingReportOperations() {
...
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({
+      value: !status,
+    }),
+  }).then((resp) => {
+    if (resp && resp.ok) {
+      return resp;
+    } else {
+      throw new Error('Error setting External Feature settings');
```
Comment (azayati) [inline webapp/portlet/src/main/webapp/spaces-administration/spacesAdministrationServices.js]:
```suggestion
      throw new Error('Error when setting External Feature settings');
```

File: webapp/portlet/src/main/webapp/spaces-administration/components/ExoSpacesAdministrationManageSpaces.vue [hunk 7]
```diff
@@ -1,11 +1,36 @@
...
+              <v-list-item-subtitle class="text-sub-title text-capitalize font-italic">
+                <div>
+                  {{ $t('social.spaces.administration.manageSpaces.toDisableExternal') }}
+                </div>
+              </v-list-item-subtitle>
+            </v-list-item-content>
+            <v-list-item-action>
+              <v-switch
+                v-model="ExternalFeatureEnabled"
+                @change="saveExternalFeatureStatus(!ExternalFeatureEnabled)"/>
```
Comment (azayati) [inline webapp/portlet/src/main/webapp/spaces-administration/components/ExoSpacesAdministrationManageSpaces.vue]:
```suggestion
                @change="saveExternalFeatureStatus(!externalFeatureEnabled)"/>
```

File: webapp/portlet/src/main/webapp/spaces-administration/components/ExoSpacesAdministrationManageSpaces.vue [hunk 8]
```diff
@@ -216,7 +242,10 @@ export default {
     },
   },
   created() {
-    this.initSpaces();	
+    this.initSpaces();
+    spacesAdministrationServices.isExternalFeatureActive().then(status => {
+      this.ExternalFeatureEnabled = status !== null ? status.value === 'true' : true;
```
Comment (azayati) [inline webapp/portlet/src/main/webapp/spaces-administration/components/ExoSpacesAdministrationManageSpaces.vue]:
```suggestion
      this.externalFeatureEnabled= status !== null ? status.value === 'true' : true;
```

File: webapp/portlet/src/main/webapp/spaces-administration/spacesAdministrationServices.js [hunk 9]
```diff
@@ -113,3 +113,45 @@ export function getBindingReportOperations() {
...
+export function saveExternalFeatureStatus(status) {
+  return fetch(`${eXo.env.portal.context}/${eXo.env.portal.rest}/v1/settings/GLOBAL/GLOBAL/exo:externalUsers`, {
+    method: 'PUT',
+    credentials: 'include',
+    headers: {
+      Accept: 'application/json',
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({
+      value: !status,
```
Comment (azayati) [inline webapp/portlet/src/main/webapp/spaces-administration/spacesAdministrationServices.js]:
```suggestion
      value: status,
```

**Review by azayati — CHANGES_REQUESTED** [review]
See review

File: webapp/portlet/src/main/webapp/spaces-administration/components/ExoSpacesAdministrationManageSpaces.vue [hunk 10]
```diff
@@ -216,7 +216,7 @@ export default {
     },
   },
   created() {
-    this.initSpaces();	
+    this.initSpaces();
```
Comment (azayati) [inline webapp/portlet/src/main/webapp/spaces-administration/components/ExoSpacesAdministrationManageSpaces.vue]: Revert this file

File: webapp/portlet/src/main/webapp/spaces-administration/components/ExoSpacesAdministrationSpacesPermissions.vue [hunk 11]
```diff
@@ -313,10 +341,18 @@ export default {
...
       }
       this.getSettingValueSpacesAdministrators();
+    },
+    saveExternalFeatureStatus(status) {
+      spacesAdministrationServices.saveExternalFeatureStatus(status);
     }
   }
 };
 </script>
+<style>
```
Comment (azayati) [inline webapp/portlet/src/main/webapp/spaces-administration/components/ExoSpacesAdministrationSpacesPermissions.vue]: Remove this useless block

**Review by azayati — CHANGES_REQUESTED** [review]
See review

File: webapp/portlet/src/main/webapp/spaces-administration/components/ExoSpacesAdministrationManageSpaces.vue (hunk 10 above)
Comment (AzmiTouil) [reply webapp/portlet/src/main/webapp/spaces-administration/components/ExoSpacesAdministrationManageSpaces.vue]: Done!

File: webapp/portlet/src/main/webapp/spaces-administration/components/ExoSpacesAdministrationSpacesPermissions.vue (hunk 11 above)
Comment (AzmiTouil) [reply webapp/portlet/src/main/webapp/spaces-administration/components/ExoSpacesAdministrationSpacesPermissions.vue]: Done!

**Review by azayati — APPROVED** [review]
LGTM after QA validation

File: webapp/portlet/src/main/resources/locale/portlet/social/SpacesAdministrationPortlet_fr.properties [hunk 12]
```diff
@@ -55,6 +55,8 @@ social.spaces.administration.permissions.descriptionCreateSpace=Possibilit\u00E9
 social.spaces.administration.permissions.manageSpaces=G\u00E9rer les espaces
 social.spaces.administration.permissions.descriptionManageSpaces=Possibilit\u00E9 de modifier les espaces
 social.spaces.administration.permissions.noAssignment=permission non assign\u00E9e
+social.spaces.administration.permissions.inviteNewExternal=Inviter un nouveau externe
+social.spaces.administration.permissions.toDisableExternal=Pour d\u00E9sactiver la capacit\u00E9 d'inviter de nouveaux membres externes.
```
Comment (azayati) [inline webapp/portlet/src/main/resources/locale/portlet/social/SpacesAdministrationPortlet_fr.properties]:
```suggestion
social.spaces.administration.permissions.enableDisableExternal=Pour activer/d\u00E9sactiver la capacit\u00E9 d'inviter de nouveaux membres externes.
```

File: webapp/portlet/src/main/resources/locale/portlet/social/SpacesAdministrationPortlet_en.properties [hunk 13]
```diff
@@ -55,6 +55,8 @@ social.spaces.administration.permissions.descriptionCreateSpace=Ability to creat
 social.spaces.administration.permissions.manageSpaces=Manage spaces
 social.spaces.administration.permissions.descriptionManageSpaces=Ability to edit spaces
 social.spaces.administration.permissions.noAssignment=No assignment
+social.spaces.administration.permissions.inviteNewExternal=Invite new external
+social.spaces.administration.permissions.toDisableExternal=To disable the capacity to invite any new external members.
```
Comment (azayati) [inline webapp/portlet/src/main/resources/locale/portlet/social/SpacesAdministrationPortlet_en.properties]:
```suggestion
social.spaces.administration.permissions.enableDisableExternal=To enable/disable the capacity to invite any new external members.
```

File: webapp/portlet/src/main/webapp/spaces-administration/components/ExoSpacesAdministrationSpacesPermissions.vue [hunk 14]
```diff
@@ -1,5 +1,29 @@
...
+                <v-col class="col-2 inviteExternalSwitcher">
+                  <v-switch
+                    v-model="externalFeatureEnabled"
+                    class="mt-5"
+                    @change="saveExternalFeatureStatus(externalFeatureEnabled)"/>
+                </v-col>
+              </v-row>
+            </v-list-item-title>
+            <v-list-item-subtitle class="text-sub-title font-italic">
+              {{ $t('social.spaces.administration.permissions.toDisableExternal') }}
```
Comment (azayati) [inline webapp/portlet/src/main/webapp/spaces-administration/components/ExoSpacesAdministrationSpacesPermissions.vue]:
```suggestion
              {{ $t('social.spaces.administration.permissions.enableDisableExternal') }}
```

**Review by azayati — CHANGES_REQUESTED** [review]
See review

File: webapp/portlet/src/main/webapp/spaces-administration/components/ExoSpacesAdministrationSpacesPermissions.vue [hunk 15]
```diff
@@ -114,14 +117,16 @@ export default {
...
+      externalFeatureEnabled: null,
+      statusLoaded: false,
     };
   },
   created() {
     this.getSettingValueCreateSpace();
     this.getSettingValueSpacesAdministrators();
     spacesAdministrationServices.isExternalFeatureActive().then(status => {
       this.externalFeatureEnabled = status !== null ? status.value === 'true' : true;
+      this.statusLoaded = true;
```
Comment (azayati) [inline webapp/portlet/src/main/webapp/spaces-administration/components/ExoSpacesAdministrationSpacesPermissions.vue]: externalFeatureStatusLoaded

File: webapp/portlet/src/main/webapp/spaces-administration/components/ExoSpacesAdministrationSpacesPermissions.vue [hunk 16]
```diff
@@ -114,14 +117,16 @@ export default {
...
       displayNoAssignmentAdministrators: false,
-      externalFeatureEnabled: true
+      externalFeatureEnabled: null,
+      statusLoaded: false,
     };
   },
   created() {
     this.getSettingValueCreateSpace();
     this.getSettingValueSpacesAdministrators();
     spacesAdministrationServices.isExternalFeatureActive().then(status => {
```
Comment (azayati) [inline webapp/portlet/src/main/webapp/spaces-administration/components/ExoSpacesAdministrationSpacesPermissions.vue]: Put this into a method getExternalFeatureStatus

**Review by azayati — CHANGES_REQUESTED** [review]
See review

File: webapp/portlet/src/main/resources/locale/portlet/social/SpacesAdministrationPortlet_en.properties [hunk 17]
```diff
@@ -55,6 +55,8 @@ social.spaces.administration.permissions.descriptionCreateSpace=Ability to creat
 social.spaces.administration.permissions.manageSpaces=Manage spaces
 social.spaces.administration.permissions.descriptionManageSpaces=Ability to edit spaces
 social.spaces.administration.permissions.noAssignment=No assignment
+social.spaces.administration.permissions.inviteNewExternal=Invite new external
+social.spaces.administration.permissions.enableDisableExternal=To enable/disable the capacity to invite any new external members.
```
Comment (azayati) [inline webapp/portlet/src/main/resources/locale/portlet/social/SpacesAdministrationPortlet_en.properties]:
```suggestion
social.spaces.administration.permissions.enableDisableExternal=To enable or disable the capacity to invite any new external members.
```

File: webapp/portlet/src/main/resources/locale/portlet/social/SpacesAdministrationPortlet_fr.properties [hunk 18]
```diff
@@ -55,6 +55,8 @@ social.spaces.administration.permissions.descriptionCreateSpace=Possibilit\u00E9
 social.spaces.administration.permissions.manageSpaces=G\u00E9rer les espaces
 social.spaces.administration.permissions.descriptionManageSpaces=Possibilit\u00E9 de modifier les espaces
 social.spaces.administration.permissions.noAssignment=permission non assign\u00E9e
+social.spaces.administration.permissions.inviteNewExternal=Inviter un nouveau externe
+social.spaces.administration.permissions.enableDisableExternal=Pour activer/d\u00E9sactiver la capacit\u00E9 d'inviter de nouveaux membres externes.
```
Comment (azayati) [inline webapp/portlet/src/main/resources/locale/portlet/social/SpacesAdministrationPortlet_fr.properties]:
```suggestion
social.spaces.administration.permissions.enableDisableExternal=Pour activer ou d\u00E9sactiver la capacit\u00E9 d'inviter de nouveaux membres externes.
```

**Review by azayati — CHANGES_REQUESTED** [review]
See review

**Review by azayati — APPROVED** [review]
LGTM after QA validation

**Pull request merged**